zookeeper-metrics-providers/zookeeper-prometheus-metrics/src/test/java/org/apache/zookeeper/metrics/prometheus/PrometheusMetricsProviderTest.java

```diff
@@ -76,31 +77,51 @@ public void testCounters() throws Exception {
         counter.add(10);
         int[] count = {0};
         provider.dump((k, v) -> {
-            assertEquals("cc", k);
-            assertEquals(10, ((Number) v).intValue());
             count[0]++;
+            int value = ((Number) v).intValue();
+            switch (k) {
+                case "cc_total":
```

Comment:
so with this upgrade we are changing the metrics that we are providing ?

can you please explain in the description of the PR and/or in JIRA ?

this may be a significant impact for people who created downstream processing dashboards/alerts

Reply:
Yes, Prometheus switched to OpenMetrics data model in 0.10.0 release. These means the following. 

1.  all Counter samples now have a _total suffix
2. Counter, Summary, and Histogram metrics now expose a _created sample with when their child was created

More details can be found in the Prometheus change logs https://github.com/prometheus/client_java/releases

Yes, the downstream processing dashboard/alerts need to make the change accordingly when they upgrade ZK version.